doc/presentations/golangmtl-march-2016/presentation.md

```diff
@@ -0,0 +1,292 @@
+# go-lua: A Lua 5.2 VM in Pure Go
+### Francis Bogsanyi
+#### Production Engineering - Shopify
+
+---
+
+# What is go-lua?
+
+---
+
+# What is ~~go-~~Lua?
+
+- Embeddable scripting language
+  - http://www.lua.org/
+  - Reference implementation around 20 kLOC of C
+- Used extensively in:
+  - **Games**: describe levels, data, game logic & AI
+  - **Nginx**: load balancing, SSL cert. management, routing, etc.
+  - **Redis**: "stored procedures"
+
+^ Assume some knowledge of Go, so let's focus on Lua.
+
+^ Reference implementation is one of the fastest scripting language interpreters. `luajit` is even faster.
+
+---
+
+# What is go-lua?
+
+- Several bindings to C Lua already exist for Go
+- go-lua is a manual port of C Lua to Go
+
+---
+
+# Origin myth
+
+- Conan load generator
+  - Written & scripted in Ruby, hosted in Heroku, scaled manually
+  - "**go**nan" rewrite - popular for pair programming interviews
+- Genghis load generator
+  - Written in Go, Heroku scheduler, EC2 workers
+  - Scales to 20M RPM against Shopify
```

Comment:
sharing this outside shopify you might want to convert RPM to RPS, since RPM is something NewRelic made up and not well used outside NewRelic customers.